python/grass/script/raster.py

```diff
@@ -261,3 +269,139 @@ def raster_what(map, coord, env=None, localized=False):
             data.append(tmp_dict)
 
     return data
+
+
+class MaskManager:
+    """Context manager for setting and managing 2D raster mask.
+
+    The context manager makes it possible to have custom mask for the current process.
+    In the following example, we set the mask using _r.mask_ which creates a new
+    raster which represents the mask. The mask is deactivated at the end of the
+    context by the context manager and the raster is removed.
+
+    >>> with gs.MaskManager():
+    ...     gs.run_command("r.mask", raster="state_boundary")
+    ...     gs.parse_command("r.univar", map="elevation", format="json")
+
+    The _mask_name_ can be a name of an existing raster map and in that case,
+    that raster map is used as mask right away. If the raster map does not exist,
+    the name will be used for the mask once it is created (with _r.mask_).
+
+    The following example uses an existing raster map directly as the mask.
+    The mask is disabled at the end of the context, but the raster map is not
+    removed.
+
+    >>> with gs.MaskManager(mask_name="state_boundary"):
+    ...     gs.parse_command("r.univar", map="elevation", format="json")
+
+    Note the difference between using the name of an existing raster map directly
+    and using *r.mask* to create a new mask. Both zeros and NULL values are used
+    to represent mask away the input data resulting in NULL cells, while *r.mask*
```

Comment:
I tried to fix this:
```suggestion
    to represent mask resulting in NULL cells, while *r.mask*
```